### ISSUE TITLE
Blocks project loading

### ISSUE THREAD

**Issue by OzBob** [justcla/HotTips#8]
when prompting to 'restart from the begining', all project loading(?), or at least all other UI elements are blocked, after click 'cancel' then the 'project loading' dialog appears.

I would suggest 
 - automatically start from the begging.
 - add tips interactiely 'onload' for all the current users's keyboard shortcuts
 - add all the tips from all the 'Hot' extensions and links to the extension gallery to install them

**Comment by justcla** [comment]
Thanks @OzBob 

Ooops. That's pretty bad of me. I'm blocking the UI. I guess I expected this to rarely happen (once we've loaded over 200 tips). But it's be showing up frequently while the extension is in the early stages.
The only reason I would not automatically restart is that I wanted the user to confirm they will be losing their tip history. The only consequence is the "Previous" button won't reload their olds tips any more. Perhaps we just accept that and do it. Then again... this should only happen after a couple of hundred loads (in theory), so should be a minor inconvenience.

**Comment by justcla** [comment]
I think there would be far too many shortcuts in the current user's profile. We would be giving them tips they don't want.
A golden principle I was working with was: The tips must be so good the user wants another one.
Otherwise, they will probably disable the feature after one or two tips.

So I want to hand-pluck the best and give them the highest priority.

**Comment by justcla** [comment]
Thanks for the support about the "Hot" extensions.

I can't suggest the tips for them if they don't have them installed, and HotTips isn't that smart yet. It uses static tips.
But perhaps, at the very least, I could slide in some cheeky advertisements for some other extensions that I really think users should know about (like Hot Settings, Hot Source and Whack Whack Terminal)

I could design HotTips so other extensions can register a TipProvider (in fact, I already designed it with that in mind).

**Comment by OzBob** [comment]
Yes, determining ahead of time what tips a user would find 'so good they want another' will be a challenge, and I can't think of a way to work which tips are 'tips they don't want'.

@kuhlenhuth might be able to supply the top 50 tips that are commonly under-used in Visual Studio, and initially they could eb turned into static html tips.

TO get away from teh static html per version of the extension. What about a HttpClient that pulls that latest tips from the github directory (liked cached webpages with an updated eTag). This might not be an issue if everyone has 'autoupdate extensions' turned on.

The Previous button could (if at tips[0]) go to the last tip in the list, just have an endless loop of tips.

**Comment by justcla** [comment]
To decide the best tips, I figured I'd just use my best judgement. But then, if we add telemetry to all the buttons and the "Stop showing tips" button (yet to come), then we should be able to see if there are any tips that consistently drive users to see "Next" or "No more".

The "More like this" button can help us learn which Group of tips is most popular.

So - two things we need:
1. Stop showing tips option.
2. Telemetry on all buttons/options/open/close.

**Comment by justcla** [comment]
Good point - I think Kasey will be keen to throw some tips my way. I'll check in with her.

**Comment by justcla** [comment]
I seriously considered the Http fetch option for the tips, but I wanted it to be super fast and thought this would be the best/most reliable.
However, I've discovered that there is a serious lag when trying to render the Web browser for the first time.

I do expect people to be fetching automatic updates, and figured that removes the need to fetch live from GitHub.

I don't think it makes sense to go "Previous" to an item you've never seen before.
Also, it's not a rolling model of tips. The way I designed it, every time you hit next it works out what tip is the next best to show based on history (which changes if they hit "More like this") and the current tip list (which could change with every update).
